scrapy/settings/__init__.py

```diff
@@ -93,7 +97,12 @@ def __getitem__(self, opt_name):
     def __contains__(self, name):
         return name in self.attributes
 
-    def get(self, name, default=None):
+    def _get_default(self, value, default, required):
+        if not required and value is _NOT_SET:
+            return default
+        return value
+
+    def get(self, name, default=_NOT_SET, required=False):
```

Comment:
What do you think about getting `required` a kwarg-only argument, to discourage this?
`settings.getbool("foo", True, False)`